lib/datadog/core/environment/process.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+require_relative 'ext'
+
+module Datadog
+  module Core
+    module Environment
+      # Retrieves process level information
+      # TODO: Add memoization
+      module Process
+        module_function
+
+        def entrypoint_workdir
+          File.basename(Dir.pwd)
+        end
+
+        def entrypoint_type
+          Core::Environment::Ext::PROCESS_TYPE
+        end
+
+        def entrypoint_name
+          File.basename($0.to_s)
+        end
+
+        def entrypoint_basedir
+          current_basedir = File.expand_path(File.dirname($0.to_s))
+          normalized_basedir = current_basedir.tr(File::SEPARATOR, '/')
+          normalized_basedir.sub(%r{^/}, '')
```

Comment:
Not sure if we'll keep this code like this, but in case we do:
```suggestion
          normalized_basedir.delete_prefix!('/')
```